.github/workflows/deploy-env.yml

```diff
@@ -0,0 +1,56 @@
+name: Deploy
+on:
+  push:
+    branches:
+      - master
+jobs:
+  deploy:
+    name: Deploy to Dev
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.13
+      - name: Install pnpm
+        run: npm install -g pnpm
+      - name: Install dependencies
+        run: ./scripts/install.sh
+      - name: Build all packages
+        run: ./scripts/build-all-packages.sh
+      - name: Deploy
+        env:
+          STAGE_NAME: e2etest
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID}}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        run: |
+          cp ./main/end-to-end-tests/e2eGitHubConfig.yml ./main/config/settings/${STAGE_NAME}.yml
+          ./scripts/environment-deploy.sh ${STAGE_NAME}
+  cypress-test:
+    name: Cypress test
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - name: Install pnpm
+        run: |
+          npm install -g pnpm
+          apt-get install libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
+      - name: Install dependencies
+        run: pnpm install
+        working-directory: main/end-to-end-tests
+      - name: Run cypress test
+        run: pnpm run cypress:run-tests:github
+        working-directory: main/end-to-end-tests
+        env:
+          CYPRESS_BASE_URL: ${{ secrets.CYPRESS_BASE_URL}}
+          cypress_researcherPassword: ${{ secrets.CYPRESS_RESEARCHERPASSWORD}}
```

Comment:
GitHub would capitalize all the secret value names, regardless of the input, so I updated the secret mapping to all cap like `CYPRESS_RESEARCHERPASSWORD`. I'll change it to `CYPRESS_RESEARCHER_PASSWORD`. 

The `CYPRESS_` header was added since cypress only passes in OS env variable with the `CYPRESS_` or `cypress_` header, and it strip the `cypress_` header before passing it on to the test. I'll add a comment to explain the `cypress_` header. 

+1 on having userName and password both passed in through Secret.